pastas/modelcompare.py

```diff
@@ -365,10 +379,10 @@ def get_diagnostics(
         if models is None:
             models = self.models
 
-        diags = DataFrame(index=[x.name for x in models])
-        for ml in models:
+        diags = DataFrame(index=self.modelnames)
+        for i, ml in enumerate(models):
             mldiag = ml.stats.diagnostics()
-            diags.loc[f"{ml.name}", mldiag.index] = mldiag[diag_col].values
+            diags.loc[self.modelnames[i], mldiag.index] = mldiag[diag_col].values
```

Comment:
This doesn't look right? If models is passed in separately, then self.modelnames doesn't necessarily match that list/order

Reply:
Fixed in patch